### DISCUSSION TITLE
Why are `is_superuser` and `is_active` in the register function?

### DISCUSSION THREAD

**Discussion by caniko** [fastapi-users/fastapi-users#1032] [Q&A]
Shouldn't these fields be kept away from the API? I am asking because I am using a swagger codegen in my stack, and I would like it to exclude these fields.

[`get_register_router`](https://github.com/fastapi-users/fastapi-users/blob/152621c679c59c046d47a83c14fd630b9d703f8a/fastapi_users/router/register.py#L10)

I understand that the `create` function has the `safe` argument; however, I don't understand why we have to do it this way. Seems like these should be two separate functions instead of one.

**Comment by caniko** [comment]
@frankie567 is there a problem? I really don't know where else to ask this question.

I have been asking questions and posting ideas for a while on this discussion forum, but ever since our [spicy exchange](https://github.com/fastapi-users/fastapi-users/discussions/977) you have been ignoring my posts. Can we keep a professional relationship? I meant everything I said on our last exchange, and I will keep your opinions in mind in our interactions from now on. I really do think our approach to collaboration are worlds apart, and that is fine. So please, let's make this work.

You are free to email me as a response.

**Comment by frankie567** [comment]
Well, I would say the main reason is legacy and conveniency. If we wanted to do everything right, we would need to have a "public" `UserRegister` without those fields and a "private" one with them. Same would go for `UserUpdate`.

The main drawback with this is that it adds even more burden to the installation/configuration path since you need to define two more schemas.

I'm not a real fan of the `safe` flag pattern, and I know it pollutes the OpenAPI schema, but I'm not sure what's the best trade-off here.

**Reply by caniko** [reply]
How about not exposing these fields to `UserRegister`, and letting the database use default values? I don't see verification and setting user to superuser fitting into registration; this should be done after registration, IMHO.
